lib/LDA_CTR.py

```diff
@@ -0,0 +1,123 @@
+LDA.
+"""
+from lib.content_based import ContentBased
+from overrides import overrides
+from sklearn.decomposition import LatentDirichletAllocation
+import numpy
+from numpy.linalg import solve
+
+
+class LDACTRRecommender(ContentBased):
+    """
+    LDA Recommender, a content based recommender that uses LDA.
+    """
+    def __init__(self, initializer, evaluator, hyperparameters, options,
+                 verbose=False, load_matrices=True, dump_matrices=True):
+        """
+        Constructor of Latent Dirichilet allocation's processor.
+
+        :param ModelInitializer initializer: A model initializer.
+        :param Evaluator evaluator: An evaluator of recommender and holder of input.
+        :param dict hyperparameters: A dictionary of the hyperparameters.
+        :param dict options: A dictionary of the run options.
+        :param boolean verbose: A flag for printing while computing.
+        :param boolean load_matrices: A flag for reinitializing the matrices.
+        :param boolean dump_matrices: A flag for saving the matrices.
+        """
+        super(LDACTRRecommender, self).__init__(initializer, evaluator, hyperparameters, options,
+                                             verbose, load_matrices, dump_matrices)
+
+    def train_k_fold(self):
+        all_errors = []
+        for current_k in range(self.k_folds):
+            self.set_data(*self.evaluator.get_fold(current_k, self.fold_test_indices))
+            self.hyperparameters['fold'] = current_k
+            self.train_one_fold(False)
+            all_errors.append(self.get_evaluation_report())
+        return numpy.mean(all_errors, axis=0)
+
+    @overrides
+    def train_one_fold(self, return_report=True):
+        """
+        Train one fold for n_iter iterations from scratch.
+
+        :param bool return_report: A flag to decide if we should return the evaluation report.
+        """
+        # Try to read from file.
+        matrix_found = False
+        if self._load_matrices is True:
+            matrix_shape = (self.n_items, self.n_factors)
+            matrix_found, matrix = self.initializer.load_matrix(self.hyperparameters, 'document_distribution_lda',
+                                                                matrix_shape)
+            self.document_distribution = matrix
+            if self._verbose and matrix_found:
+                print("Document distribution was set from file, will not train.")
+        if matrix_found is False:
+            if self._verbose and self._load_matrices:
+                print("Document distribution file was not found, will train LDA.")
+            self._train()
+            if self._dump_matrices:
+                self.initializer.save_matrix(self.document_distribution, 'document_distribution_lda')
+        self.user_vecs = numpy.random.random((self.n_users, self.n_factors))
+        self.training_data_mask = self.evaluator.get_fold_training_data_mask()
```

Comment:
Should this function get_fold_training_data_mask be included in the pull request?